src/Xamarin.Android.Tools.ApiXmlAdjuster/JavaApi.XmlModel.cs

```diff
@@ -24,15 +26,61 @@ public JavaPackage (JavaApi? parent)
 		{
 			Parent = parent;
 			
-			Types = new List<JavaType> ();
+			Types = new Dictionary<string, List<JavaType>> (StringComparer.OrdinalIgnoreCase);
 		}
 		
 		public  JavaApi?                Parent              { get; private set; }
 
 		public  string?                 Name                { get; set; }
 		public  string?                 JniName             { get; set; }
-		public  IList<JavaType>         Types               { get; set; }
-		
+
+		// Yes, there can be multiple types with the same *Java* name.
+		// For example:
+		// - MyInterface
+		// - MyInterfaceConsts
+		// It's debatable whether we handle this "properly", as most callers just
+		// do `First ()`, but it's been working for years so I'm not changing it.
+		// Exposes an IReadOnlyDictionary so caller cannot bypass our AddType/RemoveType code.
+		public  IReadOnlyDictionary<string, List<JavaType>>   Types               { get; }
```

Comment:
…alternatively, should this be `Set<JavaType>` instead of `List<JavaType>`?  Does it make sense to allow for the *same instance* to be added multiple times?

Reply:
The same "compatibility"/"diff minimization" question/concern applies to `Types` vs. `AllTypes`: I suspect that if `Types` became an `IEnumerable<JavaType>` property and `AllTypes` were the "Dictionary-like" property, fewer changes would be required.